openfl/component/aggregator/aggregator.py

```diff
@@ -768,12 +765,12 @@ def _compute_validation_related_task_metrics(self, task_name):
         # collaborator in our subset, and apply the correct
         # transformations to the tensorkey to resolve the aggregated
         # tensor for that round
-        task_agg_function = self.assigner.get_aggregation_type_for_task(task_name)
-        task_key = TaskResultKey(task_name, collaborators_for_task[0], self.round_number)
+        task_agg_function = self.assigner.get_aggregation_type_for_task(task.function_name)
```

Comment:
Aggregation function should be determined by task instead of function name

Reply:
What I meant was this should be changed to:
```task_agg_function = self.assigner.get_aggregation_type_for_task(task.name)```

Different instantiations of the same validation / train function could have different aggregation methods applied